List Item with Large Words Overflowing RichEditor

## Description

List Items overflow rich editor when there are large words. It looks like they do not apply the `word-wrap: break-word` styling.

### Steps to reproduce

1. Add a list to the RichEditor (either ul or ol is fine).
2. Type a long unbroken word.

### Expected result

The list-item should apply the `word-wrap: break-word` styling.

### Actual result

The list-item overflows off the RichEditor.

<img width="1264" alt="image" src="https://user-images.githubusercontent.com/8443215/193995944-7b8bbadc-7ba5-4932-ba68-7d1921a17b99.png">


### Environment

#### Tinacms Package Versions
Sorry, couldn't get `npm ls | grep tinacms | grep -v deduped` to work.
<!--
Run the following command to get the list of tinacms-package versions in your node_modules:

    npm ls | grep tinacms | grep -v deduped
-->

```text
pm ERR! code ELSPROBLEMS
npm ERR! extraneous: @eslint/eslintrc@0.4.3 /Users/davidtaing/Desktop/opensource/tinacms/node_modules/@eslint/eslintrc
npm ERR! extraneous: @next/eslint-plugin-next@12.0.3 /Users/davidtaing/Desktop/opensource/tinacms/node_modules/@next/eslint-plugin-next
npm ERR! extraneous: @rushstack/eslint-patch@1.1.3 /Users/davidtaing/Desktop/opensource/tinacms/node_modules/@rushstack/eslint-patch
npm ERR! missing: @tinacms/app@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/app, required by root@
npm ERR! missing: @tinacms/auth@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/auth, required by root@
npm ERR! missing: @tinacms/cli@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/cli, required by root@
npm ERR! missing: @tinacms/datalayer@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/datalayer, required by root@
npm ERR! missing: @tinacms/graphql@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/graphql, required by root@
npm ERR! missing: @tinacms/mdx@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/mdx, required by root@
npm ERR! missing: @tinacms/metrics@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/metrics, required by root@
npm ERR! missing: @tinacms/schema-tools@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/schema-tools, required by root@
npm ERR! missing: @tinacms/scripts@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/scripts, required by root@
npm ERR! missing: @tinacms/sharedctx@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/sharedctx, required by root@
npm ERR! missing: @tinacms/starter@file:/Users/davidtaing/Desktop/opensource/tinacms/experimental-examples/tina-cloud-starter, required by root@
npm ERR! missing: @tinacms/toolkit@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/toolkit, required by root@
npm ERR! missing: @tinacms/webpack-helpers@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/webpack-helpers, required by root@
npm ERR! extraneous: @typescript-eslint/scope-manager@5.15.0 /Users/davidtaing/Desktop/opensource/tinacms/node_modules/@typescript-eslint/scope-manager
npm ERR! extraneous: @typescript-eslint/types@5.15.0 /Users/davidtaing/Desktop/opensource/tinacms/node_modules/@typescript-eslint/types
npm ERR! extraneous: @typescript-eslint/visitor-keys@5.15.0 /Users/davidtaing/Desktop/opensource/tinacms/node_modules/@typescript-eslint/visitor-keys
npm ERR! missing: create-tina-app@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/create-tina-app, required by root@
npm ERR! extraneous: eslint-config-next@12.0.3 /Users/davidtaing/Desktop/opensource/tinacms/node_modules/eslint-config-next
npm ERR! extraneous: eslint-import-resolver-node@0.3.6 /Users/davidtaing/Desktop/opensource/tinacms/node_modules/eslint-import-resolver-node
npm ERR! extraneous: eslint-module-utils@2.7.3 /Users/davidtaing/Desktop/opensource/tinacms/node_modules/eslint-module-utils
npm ERR! extraneous: eslint-utils@1.4.3 /Users/davidtaing/Desktop/opensource/tinacms/node_modules/eslint-utils
npm ERR! missing: kitchen-sink-starter@file:/Users/davidtaing/Desktop/opensource/tinacms/experimental-examples/kitchen-sink, required by root@
npm ERR! missing: kitchen-sink@file:/Users/davidtaing/Desktop/opensource/tinacms/experimental-examples/unit-test-example, required by root@
npm ERR! missing: next-tinacms-cloudinary@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/next-tinacms-cloudinary, required by root@
npm ERR! missing: next-tinacms-dos@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/next-tinacms-dos, required by root@
npm ERR! missing: next-tinacms-s3@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/next-tinacms-s3, required by root@
npm ERR! missing: starter-basic-iframe@file:/Users/davidtaing/Desktop/opensource/tinacms/examples/basic-iframe, required by root@
npm ERR! missing: starter-basic@file:/Users/davidtaing/Desktop/opensource/tinacms/examples/basic, required by root@
npm ERR! missing: starter-empty@file:/Users/davidtaing/Desktop/opensource/tinacms/examples/empty, required by root@
npm ERR! missing: tinacms@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/tinacms, required by root@

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/davidtaing/.npm/_logs/2022-10-05T06_43_09_904Z-debug-0.log
root@ /Users/davidtaing/Desktop/opensource/tinacms
├── UNMET DEPENDENCY @tinacms/app@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/app
├── UNMET DEPENDENCY @tinacms/auth@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/auth
├── UNMET DEPENDENCY @tinacms/cli@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/cli
├── UNMET DEPENDENCY @tinacms/datalayer@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/datalayer
├── UNMET DEPENDENCY @tinacms/graphql@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/graphql
├── UNMET DEPENDENCY @tinacms/mdx@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/mdx
├── UNMET DEPENDENCY @tinacms/metrics@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/metrics
├── UNMET DEPENDENCY @tinacms/schema-tools@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/schema-tools
├── UNMET DEPENDENCY @tinacms/scripts@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/scripts
├── UNMET DEPENDENCY @tinacms/sharedctx@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/sharedctx
├── UNMET DEPENDENCY @tinacms/starter@file:/Users/davidtaing/Desktop/opensource/tinacms/experimental-examples/tina-cloud-starter
├── UNMET DEPENDENCY @tinacms/toolkit@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/toolkit
├── UNMET DEPENDENCY @tinacms/webpack-helpers@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/@tinacms/webpack-helpers
├── UNMET DEPENDENCY create-tina-app@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/create-tina-app
├── UNMET DEPENDENCY kitchen-sink-starter@file:/Users/davidtaing/Desktop/opensource/tinacms/experimental-examples/kitchen-sink
├── UNMET DEPENDENCY kitchen-sink@file:/Users/davidtaing/Desktop/opensource/tinacms/experimental-examples/unit-test-example
├── UNMET DEPENDENCY next-tinacms-cloudinary@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/next-tinacms-cloudinary
├── UNMET DEPENDENCY next-tinacms-dos@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/next-tinacms-dos
├── UNMET DEPENDENCY next-tinacms-s3@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/next-tinacms-s3
├── UNMET DEPENDENCY starter-basic-iframe@file:/Users/davidtaing/Desktop/opensource/tinacms/examples/basic-iframe
├── UNMET DEPENDENCY starter-basic@file:/Users/davidtaing/Desktop/opensource/tinacms/examples/basic
├── UNMET DEPENDENCY starter-empty@file:/Users/davidtaing/Desktop/opensource/tinacms/examples/empty
├── UNMET DEPENDENCY tinacms@file:/Users/davidtaing/Desktop/opensource/tinacms/packages/tinacms
```

#### Tools

| Tool    | Version |
| ------- | ------- |
| browser |  Google Chrome - Version 105.0.5195.125 (Official Build) (arm64) |
| os      |  MacOS -  Version 12.0.1   |
| node    | v16.15.0 |
| npm     | 8.15.0 |

--- a/.changeset/soft-forks-exist.md
+++ b/.changeset/soft-forks-exist.md
@@ -1,0 +1,11 @@
+---
+'@tinacms/toolkit': minor
+---
+
+Fix #3253 bullet/numbered list items overflow out of text fields
+
+# Change
+
+- Add `w-full` class to list item content to prevent text overflow to the right of sidebar as shown in #3253.
+- Also change `<ol></ol>` element class from `pl-2` to `pl-4` to fix numbered list's marker overflow to the left of text field.
+

--- a/packages/@tinacms/toolkit/src/packages/fields/plugins/MdxFieldPlugin/plate/plugins/ui/components.tsx
+++ b/packages/@tinacms/toolkit/src/packages/fields/plugins/MdxFieldPlugin/plate/plugins/ui/components.tsx
@@ -205,7 +205,7 @@ export const components = () => {
         className={classNames(
           blockClasses,
           className,
-          'mb-4 pl-2 list-decimal list-inside last:mb-0'
+          'mb-4 pl-4 list-decimal list-inside last:mb-0'
         )}
         {...attributes}
         {...props}
@@ -222,7 +222,7 @@ export const components = () => {
     [ELEMENT_LIC]: ({ attributes, editor, element, className, ...props }) => (
       <span
         // without a min-width the cursor is hidden when the list is empty
-        className={classNames(className, 'inline-block align-top mb-2')}
+        className={classNames(className, 'w-full inline-block align-top mb-2')}
         {...attributes}
         {...props}
       />

